zeq/connection/broker.h

```diff
@@ -0,0 +1,56 @@
+
+/* Copyright (c) 2014-2015, Human Brain Project
+ *                          Stefan.Eilemann@epfl.ch
+ */
+
+#ifndef ZEQ_CONNECTION_BROKER_H
+#define ZEQ_CONNECTION_BROKER_H
+
+#include <zeq/receiver.h> // base class
+#include <lunchbox/debug.h>
+
+namespace zeq
+{
+namespace connection
+{
+namespace detail { class Broker; }
+
+/**
+ * Brokers subscription requests for a zeq::Receiver.
+ *
+ * Example: @include tests/connection/broker.cpp
+ */
+class Broker : public Receiver
+{
+public:
+    /**
+     * Create a new subscription broker.
+     *
+     * The given receiver has to have at least the same lifetime as this
+     * broker. The receiver and broker are automatically shared.
+     *
+     * For simplicity, only a single Receiver is handled by a Broker. The
+     * implementation should be extended if multiple receivers shall be
+     * handled.
+     *
+     * @param address the zmq reply socket address to be used.
+     * @param receiver the Receiver to manage.
+     * @throw std::runtime_error when the zmq setup failed.
+     */
+    ZEQ_API Broker( const std::string& address, Receiver& receiver );
```

Comment:
I understand that the publisher is connecting to the Broker using a known IP to announce itself. What I don't get is why the broker and the subscriber need to be bound together. To me the broker should be a standalone entity (which could run between networks) responsible of introducing publishers to subscribers, possible taking care of multiple applications (mediating in the case there are collisions in the chosen schemas). I understand that the use case under consideration is simpler, but maybe Broker should be renamed to make it clear that is a simplified version of a more general concept (that would address Bernd's comment somehow). What about BasicBroker
In any case, the publisher-subscriber protocol should be forward compatible in the sense that it should designed with the general broker in mind.